manager/controllers/app/data_catalog_interface.go

```diff
@@ -16,7 +19,48 @@ import (
 // Returns:
 // - an error if happened
 // - the new asset identifier
-func (r *FybrikApplicationReconciler) RegisterAsset(catalogID string, info *app.DatasetDetails,
+func (r *FybrikApplicationReconciler) RegisterAsset(assetID string, catalogID string, info *app.DatasetDetails,
 	input *app.FybrikApplication) (string, error) {
-	return "", errors.New("unsupported feature")
+	r.Log.Trace().Msg("RegisterAsset")
+	details := datacatalog.ResourceDetails{}
+	if info.Details != nil {
+		details.Connection = info.Details.Connection
+		details.DataFormat = info.Details.Format
+	}
+
+	geo := ""
+	if info.ResourceMetadata != nil {
```

Comment:
if info.ResourceMetadata is defined, why not to take it entirely?

Reply:
Fixed. Thanks